src/blockchain_db/blockchain_db.cpp

```diff
@@ -177,19 +177,29 @@ void BlockchainDB::add_transaction(const crypto::hash& blk_hash, const transacti
   // we need the index
   for (uint64_t i = 0; i < tx.vout.size(); ++i)
   {
+    uint64_t unlock_time = 0;
+    if (tx.version > 2)
```

Comment:
@Doy-lee and I decided to use the TX version to denote deregistration TXs and fucked up the versioning. See [here](https://github.com/loki-project/loki/blob/dev/src/cryptonote_core/blockchain.cpp#L2657) and elsewhere in check_tx_inputs for examples.

@Doy-lee Probably we should add a field to the TX is_deregistration, now that we can see how to do this.

@tewinget what do you think?

Reply:
that's one way to do it; the other would be a flag in tx_extra.  Either way is probably better than a specific tx version I guess, since checks against tx version tend to be "if > version X".

Reply:
Cool .. so I guess before accepting this in I'll have to switch the our deregister TX's to work using a bool flag.

Reply:
@Doy-lee Maybe you can do a pull-request to this branch?